react/features/toolbar/reducer.js

```diff
@@ -0,0 +1,101 @@
+/* @flow */
+
+import { ReducerRegistry } from '../base/redux';
+
+import {
+    SET_ALWAYS_VISIBLE_TOOLBAR,
+    SET_SUBJECT_SLIDE_IN,
+    SET_TOOLBAR_BUTTON,
+    SET_TOOLBAR_IS_HOVERED,
+    SET_TOOLBAR_VISIBILITY
+} from './actionTypes';
+import { getDefaultToolbarButtons } from './functions';
+
+/**
+ * Returns initial state for toolbar's part of Redux store.
+ *
+ * @returns {{
+ *      primaryToolbarButtons: Map,
+ *      secondaryToolbarButtons: Map
+ * }}
+ * @private
+ */
+function _getInitialState() {
+    return {
+        ...getDefaultToolbarButtons(),
+
+        alwaysVisible: false,
+        isHovered: false,
+        subjectSlideIn: false,
+        visible: true
+    };
+}
+
+ReducerRegistry.register('features/toolbar', (
+        state: Object = _getInitialState(),
+        action: Object) => {
+    switch (action.type) {
+    case SET_ALWAYS_VISIBLE_TOOLBAR:
+        return {
+            ...state,
+            alwaysVisible: action.alwaysVisible
+        };
+
+    case SET_SUBJECT_SLIDE_IN:
+        return {
+            ...state,
+            subjectSlideIn: action.subjectSlideIn
+        };
+
+    case SET_TOOLBAR_BUTTON:
+        return _setButton(state, action);
+
+    case SET_TOOLBAR_IS_HOVERED:
+        return {
+            ...state,
+            isHovered: action.isHovered
+        };
+
+    case SET_TOOLBAR_VISIBILITY:
+        return {
+            ...state,
+            visible: action.visible
+        };
+    }
+
+    return state;
+});
+
+/**
+ * Sets new value of the button.
+ *
+ * @param {Object} state - Redux state.
+ * @param {Object} button - Object describing toolbar button.
+ * @returns {Object}
+ * @private
+ */
+function _setButton(state, { key, button }): Object {
+    const {
+        primaryToolbarButtons,
+        secondaryToolbarButtons
+    } = state;
+    let selectedButton = primaryToolbarButtons.get(key);
+    let place = 'primaryToolbarButtons';
```

Comment:
You should dispatch two separate actions to do that. And yes, for example we can hide `desktopsharing` button from the primary toolbar and `sip` from secondary one.